src/main/java/seedu/address/logic/commands/add/AddApplicationCommand.java

```diff
@@ -67,32 +64,18 @@ public AddApplicationCommand(Index companyIndex, Index internshipIndex, Status s
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        if (!model.getTabName().equals(TabName.APPLICATION)) {
-            throw new CommandException(String.format(MESSAGE_WRONG_TAB, APPLICATION_NAME));
-        }
-
-        List<CompanyItem> lastShownCompanyList = model.getCompanyList().getFilteredItemList();
-
-        if (companyIndex.getZeroBased() >= lastShownCompanyList.size()) {
-            throw new CommandException(String.format(Messages.MESSAGE_INVALID_ITEM_DISPLAYED_INDEX, COMPANY_NAME));
-        }
-
-        CompanyItem companyItem = lastShownCompanyList.get(companyIndex.getZeroBased());
-        List<InternshipItem> internshipItemList = companyItem.getInternships();
-
-        if (internshipIndex.getZeroBased() >= internshipItemList.size()) {
-            throw new CommandException(String.format(Messages.MESSAGE_INVALID_ITEM_DISPLAYED_INDEX, INTERNSHIP_NAME));
-        }
-
-        InternshipItem internshipItem = internshipItemList.get(internshipIndex.getZeroBased());
+        InternshipItem internshipItem = InternshipAccessor.getInternship(model, companyIndex, internshipIndex);
         ApplicationItem applicationItem = new ApplicationItem(internshipItem, status, statusDate);
 
         FilterableItemList<ApplicationItem> applicationList = model.getApplicationList();
 
         if (applicationList.hasItem(applicationItem)) {
-            throw new CommandException(MESSAGE_DUPLICATE_APPLICATION);
+            throw new CommandException(String.format(MESSAGE_DUPLICATE_ITEM, APPLICATION_NAME));
         }
+
         applicationList.addItem(applicationItem);
+        model.setTabName(TabName.APPLICATION);
```

Comment:
Is this supposed to switch the tabs? It doesn't do that at the moment.

Reply:
Not sure what is the group's consensus on this, so I took the easier path first which is to switch tabs after successful command. Could it be because the command is executed unsuccessfully? Cuz I took this from @seanjyjy switch tabs command.

Reply:
okay! I tested on my add and delete commands but it doesn't do anything yet. Is there more to the switching besides setting the model state which I have missed from your implementation?  I think you may have to return the multiarg CommandResult as there is a boolean arg to indicate whether to switch ?

Reply:
Just tried using the same line of code for the add company command I'm working on and yep, even upon successful addition of a company, this line of code does not switch tabs. Will follow Isaac's observation and see if I can make it work for add company in the time being!